### PR TITLE
Page header updates to share common UI language in OSD

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
This PR updates all the pages to leverage the shared page header control updates gated behind the `[newHomePage](home:useNewHomePage)` setting.

https://github.com/user-attachments/assets/259de3c0-ffd3-480c-b053-5c1218f2d89a

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Rules/containers/CreateRule/CreateRule.tsx [hunk 1]
```diff
@@ -20,30 +20,28 @@ export interface CreateRuleProps {
...
-            <EuiLink
-              href="https://sigmahq.github.io/sigma-specification/Sigma_specification.html"
-              target="_blank"
-            >
-              Learn more in the Sigma rules specification
-            </EuiLink>
-          </EuiText>
-        </>
-      }
+      // title={
```
Comment (riysaxen-amzn) [inline public/pages/Rules/containers/CreateRule/CreateRule.tsx:29]: is this needed?

Comment (amsiglan) [reply public/pages/Rules/containers/CreateRule/CreateRule.tsx]: Thanks for catching, fixed.

File: public/pages/LogTypes/containers/LogTypes.tsx [hunk 2]
```diff
@@ -76,18 +85,25 @@ export const LogTypes: React.FC<LogTypesProps> = ({ history, notifications, data
...
+        <PageHeader appRightControls={[{ renderComponent: createLogTypeAction }]}>
+          <EuiFlexItem>
+            <EuiFlexGroup gutterSize={'s'} justifyContent={'spaceBetween'}>
+              <EuiFlexItem>
+                <EuiTitle size="m">
+                  <h1>Log types</h1>
+                </EuiTitle>
+                <EuiText size="s" color="subdued">
+                  Log types describe the data sources the detection rules are meant to be applied
+                  to.
```
Comment (AWSHurneyt) [inline public/pages/LogTypes/containers/LogTypes.tsx:99]: Nitpick:
> _Log types describe the data sources to which the detection rules are meant to be applied._

File: public/pages/Detectors/components/UpdateFieldMappings/UpdateFieldMappings.tsx [hunk 3]
```diff
@@ -166,19 +165,29 @@ export default class UpdateFieldMappings extends Component<
...
-          source.
-        </EuiText>
-
-        <EuiSpacer size={'xxl'} />
-
+        <PageHeader
+          appDescriptionControls={[
+            {
+              description: `To perform threat detections, your data source will need to be in a common schema format.
+            Rule field names are automatically mapped to the most common fields in your log data
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply public/pages/Detectors/components/UpdateFieldMappings/UpdateFieldMappings.tsx:172]: We are tracking it separately to improve the field mapping experience, so won't touch it in this PR

**Comment by amsiglan** [comment]
Passing tests
image

**Review by AWSHurneyt — APPROVED** [review]

**Review by riysaxen-amzn — APPROVED** [review]

**Pull request merged**